app/helpers/maintenance_tasks/tasks_helper.rb

```diff
@@ -118,6 +132,16 @@ def resolve_inclusion_value(task_class, parameter_name)
       return unless inclusion_validator
 
       in_option = inclusion_validator.options[:in] || inclusion_validator.options[:within]
+      in_option = in_option.call if in_option.is_a?(Proc) && in_option.arity.zero?
+
+      if in_option.is_a?(Enumerable) &&
+          !(size = in_option.size).nil? &&
+          size > 0 &&
+          size <= MaintenanceTasks.max_task_enumerable_picker_size
+
+        in_option = in_option.to_a
+      end
+
       in_option if in_option.is_a?(Array)
```

Comment:
Yeah. I can change it to also include arrays, but that's a bigger departure from the way things work today. Considering it's 1000 items by default, I don't think it matters much either way. 

The max number of items is mostly to prevent infinite arrays.

Reply:
Ok, changed it to apply to arrays too.